### ISSUE TITLE
Excessive Wi-Fi scans (background)

### ISSUE THREAD

**Issue by niqo01** [OneSignal/OneSignal-Android-SDK#433]
After releasing one signal our Excessive Wi-Fi scans (background) Android vitals rate went from 0.01% to 0.15%.
Upon doing our first push notification from onesignal to a few users this metric went crazy up, 0.34%.
Note that Android vitals bad threshold set to 0.10%.

![onesignal_excessive_wifi_scan](https://user-images.githubusercontent.com/1121251/35877119-430ad7da-0b29-11e8-8078-6674e0420803.png)

**Comment by jkasten2** [comment]
@niqo01 The SDK does not preform any wifi scans directly. If the location permission is enabled in your app OneSignal will use the `FusedLocationProvider` that could trigger wifi scans. If you have added the location permission by mistake to your `AndroidManifest.xml` removing may fix this issue.

You noted that the issue did not appear until you sent your first batch of notifications. Do note that the `onCreate()` of your `Application` class will be called when a notification is sent. If you have any location code or other SDKs that may use location in there this will get triggered as well. The [`Application.onCreate()`](https://developer.android.com/reference/android/app/Application.html#onCreate()) is called when any `Intent` starts your app including a push so OneSignal does not have control over the `onCreate` trigger.

**Comment by eriuzo** [comment]
According to [Google Play Console](https://developer.android.com/topic/performance/vitals/bg-wifi.html#android_vitals)
> Percentage of battery sessions during which users experienced more than 4 Wi-Fi scans per hour. A battery session is the period between two full charges of a device. Data is collected only when the device is off-charger.

I think this can be solved by allowing users to manually set location scanning interval, and providing default interval value of 15 minutes.

**Comment by rameshvoltella** [comment]
@jkasten2 hi i am also getting this issue, just today i saw on android vital, i am using fuse api for getting location for only one time when app is been launched first time after that no fuse api fuction called i use this for past 9 months, i check the vital, the graph progress or even started after i have integrated onesignal to the app. the graph start from april 1sth, for the other location function i havent done any change yet for past nine month till now no warning came, but after including onesignal, i got this issue. 

As you say i cant remove location permisson from my manifest. what should i do?

**Comment by rameshvoltella** [comment]
also @jkasten2  if i use .autoPromptLocation(false)  will onesignal stop to looking for location?

```
OneSignal.startInit(this)
                        .inFocusDisplaying(OneSignal.OSInFocusDisplayOption.Notification)
                        .setNotificationOpenedHandler(new OneSignalNotificationOpenedHandler(getApplicationContext()))
                        .autoPromptLocation(false)
                        .unsubscribeWhenNotificationsAreDisabled(true)
                        .init();
```


This how i am going to use will it avoid onesiganl location checking?

**Comment by jkasten2** [comment]
@rameshvoltella What version of the OneSignal SDK are you currently using in your app?

`.autoPromptLocation` is `false` by default, if set to `true` it will prompt the user for location permission on app start up if it isn't enabled already.

We are also looking into making a few changes in the next update to resolve this issue

**Comment by rameshvoltella** [comment]
Hi
i am using 3.8.3 (com.onesignal:OneSignal:3.8.3)
Two crashes and 1 bad behavior varning due to onesignal, i am sure you guys can fix it, mot users are experiencing this please give a updated version asap.
This is issue making warning. and issue #465 and #487 making huge crashes, please give priority.

**Comment by jkasten2** [comment]
This should now be fixed in the 3.8.4 update>
https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/3.8.4
